### PR TITLE
[release-1.10] Pin version for Redis, Kafka and Mongo in E2E.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Pin version for Redis, Kafka and Mongo in E2E.

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -53,7 +53,8 @@ jobs:
       DAPR_NAMESPACE: dapr-tests
       GOVER: "^1.19.2"
       # Container registry where to cache e2e test images
-      DAPR_CACHE_REGISTRY: "dapre2eacr.azurecr.io"
+      # Disabling it for release-1.1- branch.
+      DAPR_CACHE_REGISTRY: "localhost:5000/dapr"
```
Comment (ItalyPaleAle) [inline .github/workflows/kind-e2e.yaml:57]: This shouldn't be necessary - images are cached and referenced  by their sha256 so

Comment (artursouza) [reply .github/workflows/kind-e2e.yaml:57]: Yes, this was me trying an hypothesis of the caching being buggy.

**Comment by mukundansundar** [comment]
Ran the specific tests that failed in https://github.com/dapr/dapr/actions/runs/6263957070/job/17009533751?pr=6946 in AKS ... all of them pass against latest release-1.10 branch ... 
But it seems to fail in KinD here ... will investigate more.

**Comment by mukundansundar** [comment]
closing in favor of https://github.com/dapr/dapr/pull/7165

**Pull request closed without merging**